src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonService.kt

```diff
@@ -66,7 +66,12 @@ class PrisonService(
     }
 
     with(prisonInsertRecord) {
-      val prison = Prison(prisonId = prisonId, name = prisonName, active = active)
+      val prison = Prison(prisonId = prisonId, name = prisonName, active = true, male = male, female = female)
+
```

Comment:
The addAddress method name feels slightly misleading as I think it would only work correctly in your context i.e. creating a new prison within a transaction? Maybe doesn't matter here through as I know all of the address maintenance is handled by separate service.

Reply:
I totally get what you're saying.  I wasn't sure what was the best method to go down.  I based it on court register.
Adding all the addresses against the prison and then saving means they all get saved through the Jpa settings for prison ( which in turn saves its associated addresses).